tests/integration/test_membership.py

```diff
@@ -167,13 +168,15 @@ def test_remove_and_rejoin_node_with_same_id_fails(cluster, use_snapshot):
     cluster.node(cluster.leader).wait_for_num_nodes(2)
 
     logger.info("Re-add node")
-    new_node = cluster.add_node(port=port, node_id=node_id, expect_error="Failed to join cluster, check logs")
-
-    logger.info("Waiting for timeout")
-
-    # TODO: Detect the error immediately (by inspecting the logs?) instead of retrying until timeout
-    with raises(RedisRaftTimeout):
-        new_node.wait_for_node_voting()
+    try:
+        new_node = cluster.add_node(port=port, node_id=node_id, single_run=True)
+        logger.info("Waiting for timeout")
+
+        # TODO: Detect the error immediately (by inspecting the logs?) instead of retrying until timeout
+        with raises(RedisRaftTimeout):
+            new_node.wait_for_node_voting()
```

Comment:
This should be enough -- it will fail
* If `RedisRaftTimeout` is not raised
* If any other exception is raised

Reply:
this is now very changed